### PR TITLE
Fix dim order in op_permute

### PR DESCRIPTION
**Pull request by Erik-Lundell**

### CODE REVIEW

File: backends/arm/operators/op_permute.py [hunk 1]
```diff
@@ -40,8 +88,33 @@ def define_node(
             )
             return
 
+        # The permutation vector describes a permutation P in default Pytorch dim_order.
+        # For rank 4, the default dim_order NCHW.
+        # E.g. (2,3,0,1) -> permute (n,c,h,w) to (w,c,n,h)
```
Comment (digantdesai) [inline backends/arm/operators/op_permute.py:93]:
```suggestion
        # E.g. (2,3,0,1) -> permute (n,c,h,w) to (h,w,n,c)
```

File: backends/arm/operators/op_permute.py [hunk 2]
```diff
@@ -40,8 +88,33 @@ def define_node(
...
+
+            # To do this, represent P and S with permutation matrices.
+            # Matrices can handle chained transformations and inversion easily.
+            S = permutation_vector_to_matrix(output.dim_order)
+            # The inverse of a permutation matrix is its transpose.
+            S_inverse = S.transpose(1, 0)
+            P = permutation_vector_to_matrix(permutation_vector)
+
+            # The complete transformation is S * P * S_inverse.
+            transformation_matrix = S.matmul(P.matmul(S_inverse))
```
Comment (digantdesai) [inline backends/arm/operators/op_permute.py:110]: This is very cool, and I think I get what it is doing, but can you help me understand the math here.

File: backends/arm/test/ops/test_permute.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    def _test_permute_tosa_BI_pipeline(
+        self, module: torch.nn.Module, test_data: Tuple[torch.tensor]
+    ):
+        quantizer = ArmQuantizer().set_io(get_symmetric_quantization_config())
+        (
+            ArmTester(
+                module,
+                example_inputs=test_data,
+                compile_spec=common.get_tosa_compile_spec(),
```
Comment (digantdesai) [inline backends/arm/test/ops/test_permute.py:82]: can we test permute to nhwc here at least for u85?

**Review by digantdesai — APPROVED** [review]

File: backends/arm/operators/op_permute.py (hunk 1 above)
Comment (Erik-Lundell) [reply backends/arm/operators/op_permute.py:93]: Your suggestion is correct, messed this up.

File: backends/arm/operators/op_permute.py (hunk 2 above)
Comment (Erik-Lundell) [reply backends/arm/operators/op_permute.py:110]: Sure, the problem I am addressing is how a permutation vector expressed in nchw should be modified to work for nhwc. The key idea is to view the dim order as a coordinate system (the index (1,2,3,4) in nchw-coordinates is the same as (1,3,4,2) in nhwc) and to realize that both transformations between the systems and a permutation ops are linear operations that can be described by matrices.

For each nhwc-index in the incoming tensor, the full transformation of the permute op can then be described by: 
1) translate from nhwc to nchw
2) do the permutation in nchw coordinates
3) translate back to nhwc
Since all steps are permutation transformations, we can combine them into one single permutation transformation, which can then be expressed by a permutation vector - this is the vector we were looking for.

File: backends/arm/test/ops/test_permute.py (hunk 3 above)
Comment (Erik-Lundell) [reply backends/arm/test/ops/test_permute.py:82]: Good point, right now we are only testing the case when we already are in nhwc for rank 4 tensors.
